File: 0226_Invert Binary Tree/InvertBinaryTree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        if (root == nullptr) {  //若根节点为空，返回0
            return nullptr;
        }
        //左右子树交换
        TreeNode* left = invertTree(root->left);
        TreeNode* right = invertTree(root->right);
        root->left = right;  //右子树值赋值给左子树
        root->right = left;  //左子树值赋值给右子树
        return root;  //输出二叉树
    }
};

File: 0226_Invert Binary Tree/InvertBinaryTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: TreeNode) -> TreeNode:
        if not root:  #若根节点为空，返回0
            return None
        root.right,root.left = root.left,root.right #左右子树交换
        self.invertTree(root.left)  #右子树值赋值给左子树
        self.invertTree(root.right)  #左子树值赋值给右子树
        return root  #输出二叉树
